src/Marketplace/Repositories/RecipientRepository.php

```diff
@@ -2,13 +2,28 @@
 
 namespace Pagarme\Core\Marketplace\Repositories;
 
+use MundiAPILib\APIException;
+use MundiAPILib\Controllers\RecipientsController;
+use MundiAPILib\Models\GetTransferSettingsResponse;
 use Pagarme\Core\Kernel\Abstractions\AbstractDatabaseDecorator;
 use Pagarme\Core\Kernel\Abstractions\AbstractEntity;
 use Pagarme\Core\Kernel\Abstractions\AbstractRepository;
+use Pagarme\Core\Kernel\Exceptions\InvalidParamException;
 use Pagarme\Core\Kernel\ValueObjects\AbstractValidString;
+use Pagarme\Core\Marketplace\Aggregates\Recipient;
+use Pagarme\Core\Marketplace\Factories\RecipientFactory;
 
 class RecipientRepository extends AbstractRepository
 {
+    protected $controller;
+
+    public function __construct(RecipientsController $controller = null)
+    {
+        parent::__construct();
+
+        $this->controller = $controller;
```

Comment:
tem alguma razão da gente precisar receber o controller por parâmetro? Não seria melhor o próprio repositório inicializar o que ele precisa no construtor?

Reply:
O controller só é utilizado com alguns métodos em específico, não sendo necessário ser instanciado todas as vezes que o repositório for construído

Reply:
Desculpa, agora que eu notei que era um repository recebendo um controller que chama a API. Um repository não devia ter essa responsabilidade não. O repository só deveria receber um objeto de domínio pra ser persistido, quem deveria ter a responsabilidade de chamar uma API pra receber os dados, que é o que esse controller faz, é um service.

O fluxo devia ser algo assim:
1. Service chama a API e recebe a resposta
2. Service monta um objeto de domínio (esse Recipient, no caso) e passa pro repository se exibir a necessidade de persistir os dados

Sacou?

Reply:
O repositório continua recebendo o controller de API diretamente